ui/ceval/src/stockfishProtocol.js

```diff
@@ -29,55 +32,67 @@ module.exports = function(worker, opts) {
 
   var processOutput = function(text) {
     if (text.indexOf('bestmove ') === 0) {
-      emit();
       if (!stopped) stopped = m.deferred();
       stopped.resolve(true);
       return;
     }
     if (!work) return;
 
-    if (text.indexOf('currmovenumber') !== -1) return;
-    var matches = text.match(/depth (\d+) .*multipv (\d+) .*score (cp|mate) ([-\d]+) .*nps (\d+) .*pv (.+)/);
-    if (!matches) {
-      emit();
-      return;
-    }
+    var matches = text.match(EVAL_REGEX);
+    if (!matches) return;
 
-    var depth = parseInt(matches[1]);
-    if (depth < opts.minDepth) return;
-    var multiPv = parseInt(matches[2]);
-    var cp, mate;
-    if (matches[3] === 'cp') cp = parseFloat(matches[4]);
-    else mate = parseFloat(matches[4]);
-    if (work.ply % 2 === 1) {
-      if (matches[3] === 'cp') cp = -cp;
-      else mate = -mate;
-    }
+    var depth = parseInt(matches[1]),
+        multiPv = parseInt(matches[2]),
+        isMate = matches[3] === 'mate',
+        eval = parseFloat(matches[4]),
+        evalType = matches[5],
+        nodes = parseInt(matches[6]),
+        nps = parseInt(matches[7]),
+        elapsedMs = parseInt(matches[8]),
+        pv = matches[9].trim();
+
+    if (depth < opts.minDepth || !pv) return;
```

Comment:
Sorry, picking on this again. If we test `!pv`, why not just use `pv (.+)` in the regex in the first place?